Framework/Geometry/src/Objects/Rules.cpp

```diff
@@ -231,19 +236,24 @@ int Rule::makeCNFcopy(Rule *&TopRule)
           // Note:: no part of this can be memory copy
           // hence we have to play games with a second
           // gamma->clone()
-          partReplace = new Union(new Intersection(alpha, gamma),
-                                  new Intersection(beta, gamma->clone()));
+          std::unique_ptr<Rule> tmp1 =
+              Mantid::Kernel::make_unique<Intersection>(std::move(alpha),
+                                                        std::move(gamma));
+          std::unique_ptr<Rule> tmp2 =
+              Mantid::Kernel::make_unique<Intersection>(std::move(beta),
+                                                        std::move(gamma));
```

Comment:
I am not sure I understand this. Previously there was `gamma->clone()`, now we call `std::move(gamma)` twice, so the second time it should always contain a `nullptr`. I guess that is not what we want?